sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,64 +31,146 @@
+        all_params = ['df']
+        for name, value in param:
+            all_params.append('{}={}'.format(name, value))
+        func_definition = 'def _reduce_impl({}):'.format(', '.join(all_params))
+        func_lines = [func_definition]
+        for i, d in enumerate(data_args):
+            line = '  {} = hpat.hiframes.api.init_series(hpat.hiframes.pd_dataframe_ext.get_dataframe_data(df, {}))'
+            func_lines.append(line.format(d + '_S', i))
+            func_lines.append('  {} = {}.{}()'.format(d + '_O', d + '_S', name))
+        func_lines.append('  data = np.array(({},))'.format(
+            ", ".join(d + '_O' for d in data_args)))
+        func_lines.append('  index = hpat.str_arr_ext.StringArray(({},))'.format(
+            ', '.join('"{}"'.format(c) for c in saved_columns)))
+        func_lines.append('  return hpat.hiframes.api.init_series(data, index)')
+        loc_vars = {}
+        func_text = '\n'.join(func_lines)
+
+        exec(func_text, {'hpat': sdc, 'np': numpy}, loc_vars)
+        _reduce_impl = loc_vars['_reduce_impl']
+
+        return _reduce_impl
+
+    def check_type(name, df, axis=None, skipna=None, level=None, numeric_only=None, ddof=1, min_count=0):
+        ty_checker = TypeChecker('Method {}().'.format(name))
+        ty_checker.check(df, DataFrameType)
+
+        if not (isinstance(axis, types.Omitted) or axis is None):
+            ty_checker.raise_exc(axis, 'unsupported', 'axis')
+
+        if not (isinstance(skipna, (types.Omitted, types.NoneType, types.Boolean)) or skipna is None):
+            ty_checker.raise_exc(skipna, 'bool', 'skipna')
+
+        if not (isinstance(level, types.Omitted) or level is None):
+            ty_checker.raise_exc(level, 'unsupported', 'level')
+
+        if not (isinstance(numeric_only, types.Omitted) or numeric_only is None):
+            ty_checker.raise_exc(numeric_only, 'unsupported', 'numeric_only')
+
+        if not (isinstance(ddof, types.Omitted) or ddof == 1):
+            ty_checker.raise_exc(ddof, 'unsupported', 'ddof')
+
+        if not (isinstance(min_count, types.Omitted) or min_count == 0):
+            ty_checker.raise_exc(min_count, 'unsupported', 'min_count')
+
+        if not isinstance(n, (types.Integer, types.Omitted)) and n != 5:
+            ty_checker.raise_exc(n, 'integer', 'n')
+
+    @overload_method(DataFrameType, 'head')
+    def median_overload(df, n=5):
+        """
+        Pandas DataFrame method :meth:`pandas.DataFrame.head` implementation.
+        .. only:: developer
+            Test: python -m sdc.runtests sdc.tests.test_dataframe.TestDataFrame.test_head1
+        Parameters
+        -----------
+        self: :class:`pandas.DataFrame`
+            input arg
+        n: :obj:`int`, default 5
+            input arg, default 5
+        Returns
+        -------
+        :obj:`pandas.Series`
+        returns: The first n rows of the caller object.
+        """
+
+        name = 'head'
+
+        check_type(name, df, n=n)
```

Comment:
Are you sure that type-check is correct here? I don't see the named parameter 'n' in `check_type` function args